parsers/AM.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+# coding=utf-8
+
+import arrow
+import re
+import requests
+from dateutil import parser as dparser
+from dateutil import tz
+from bs4 import BeautifulSoup
+
+#URL of the power system summary: http://epso.am/poweren.htm
+#URL of the detailled SCADA-page: http://epso.am/scada.htm
+
+power_plant_mapping = {
+    'var atom': 'nuclear', # atom = 'atomnaya elektro stantsiya'
+    'var tes': 'gas',      # tes = 'termalnaya elektro stantsiya' - only gas in AM
+    'var ges': 'hydro',    # ges = 'gidro elektro stantsiya'
+    'var altern': 'hydro'  # altern = two hydro power plants according to SCADA data (middle row, right box)
+    }
+
+tie_line_mapping = {
+    'var Lalvar': 'GE',
+    'var ninoc': 'GE',
+    'var alaver': 'GE',
+    'var shin': 'AZ(AM)',
+    'var arcakh': 'AZ(AM)',
+    'var ahar': 'IR'
+    }
+
+other_variables_mapping = {
+    'var cons': 'total production', #please note, this is an error of the variable name ('cons' is actually used for total production)
+    'var peretok': 'total import/export', #positive = import, negative = export
+    'var herc2': 'frequency',       #"hertz" ;)
+    'var time2': 'timestamp',       #date is missing
+    'sparum2': 'total consumption', #please note, this is an error of the variable name (probably means "production", but is used for consumption)
+    }
+
+soup_content_variables_mapping = {
+    '[0]' : 'empty',
+    '[1]' : 'Lalvar (GE)',
+    '[2]' : 'ninoc (GE)',
+    '[3]' : 'alaver (GE)',
+    '[4]' : 'shin (AZ)',
+    '[5]' : 'arcakh (AZ)',
+    '[6]' : 'ahar (IR)',
+    '[7]' : 'cons [total production]',
+    '[8]' : ' altern',
+    '[9]' : 'atom',
+    '[10]' : 'tes',
+    '[11]' : 'ges',
+    '[12]' : 'peretok',
+    '[13]' : 'herc2',
+    '[14]' : 'time2',
+    '[15]' : 'sparum2 []',
+    '[16]' : 'empty',
+    '[17]' : 'empty',
+    }
+
+def fetch_production(zone_key='AM', session=None, target_datetime=None, logger=None):
+    r = session or requests.session()
+    url = 'http://epso.am/poweren.htm'
+    response = r.get(url)
+    response.encoding = 'utf-8'
+    html_doc = response.text
+    soup = BeautifulSoup(html_doc, 'html.parser')
+
+    data_string = soup.find(text=re.compile('var'))
+    data_split = data_string.split('\r\n')
+
+    gas_tes = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", data_split[10])
+    gas_total = float(gas_tes[0])
+
+    hydro_ges = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", data_split[11])
+    hydro_altern = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", data_split[8])
+    hydro_total = float(hydro_ges[0])+float(hydro_altern[0])
+
+    nuclear_atom = re.findall("[-+]?[.]?[\d]+(?:,\d\d\d)*[\.]?\d*(?:[eE][-+]?\d+)?", data_split[9])
+    nuclear_total = float(nuclear_atom[0])
+
+    time_string = dparser.parse(data_split[14].split()[3],fuzzy=True)
```

Comment:
We could do something like this to ensure the local time in Armenia is used, need to add a datetime import as well.
```suggestion
    yerevan = tz.gettz('Asia/Yerevan')
    date_time = dparser.parse(data_split[14].split()[3], default=datetime.now(yerevan), fuzzy=True)
```